pkg/processor/trigger/test/eventrecorder.go

```diff
@@ -114,6 +128,18 @@ func InvokeEventRecorder(suite *processorsuite.TestSuite,
 		// compare bodies
 		suite.Require().Equal(sentBodies, receivedBodies)
 
+		if postPublishChecks != nil {
+			if postPublishChecks.EnsureAckFunction != nil {
+				err = common.RetryUntilSuccessful(
+					60*time.Second,
+					2*time.Second,
+					func() bool {
+						return postPublishChecks.EnsureAckFunction(postPublishChecks.ConsumerGroup, "", postPublishChecks.ExpectedNumberOfCommittedOffsets)
+					},
+				)
+				suite.Require().NoError(err)
```

Comment:
add a comment that we retry because it can take time for the function to ack on the events